components/calendar/src/any_calendar.rs

```diff
@@ -1026,46 +877,71 @@ impl AnyCalendarKind {
         }
     }
 
-    /// Extract the calendar component from a [`Locale`]
-    ///
-    /// Returns `None` if the calendar is not specified or unknown.
-    pub fn get_for_locale(l: &Locale) -> Option<Self> {
-        l.extensions
-            .unicode
-            .keywords
-            .get(&key!("ca"))
-            .and_then(Self::get_for_bcp47_value)
-    }
-
     /// Extract the calendar component from a [`AnyCalendarPreferences`]
     ///
     /// Returns `None` if the calendar is not specified or unknown.
-    fn get_for_prefs(prefs: AnyCalendarPreferences) -> Option<Self> {
-        prefs
-            .calendar_algorithm
-            .and_then(Self::get_for_preferences_calendar)
+    pub fn from_prefs(prefs: AnyCalendarPreferences) -> Option<Self> {
```

Comment:
Issue: We don't use the name `prefs` in public API signatures.

We recently bikeshed the AnyCalendar constructor names in https://github.com/unicode-org/icu4x/issues/5843.

Also, I think this is the fn where we should resolve the calendar, once it is data-driven.

Why do we even need this fn? The `From` impls seen fine.

Reply:
> Also, I think this is the fn where we should resolve the calendar, once it is data-driven.

Calendar resolution should happen in the `AnyCalendar` constructor, as that's where we have data.

> Why do we even need this fn? The From impls seen fine.

Happy to remove